devito/ir/iet/nodes.py

```diff
@@ -1457,8 +1457,16 @@ def __init__(self, body, halo_scheme):
         self._halo_scheme = halo_scheme
 
     def __repr__(self):
-        functions = "(%s)" % ",".join(i.name for i in self.functions)
-        return "<%s%s>" % (self.__class__.__name__, functions)
+        fstrings = []
+        for f in self.functions:
+            loc_indices = set().union(*[self.halo_scheme.fmapper[f].loc_indices.values()])
```

Comment:
since this is a set, the result of the `__repr__` will be non deterministic across different python executions, that is, you may see either [t0,t1] or [t1,t0]

Reply:
Used OrderedSet